### PR TITLE
Add transforms for ultralytics yolov5 integration

### PR DESCRIPTION
**Pull request by changhiskhan**
This works locally but the testing setup is a bit of a mess. Not sure if there's a good way to do it without mucking up the rest of the tests. Maybe do it later as it's own test job or something?

1. you need to pip install yolov5 requirements (pytorch 1.7+)
2. you need to pip install yolov5 itself

### CODE REVIEW

File: python/rikai/contrib/torch/transforms/yolov5.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            for *box, conf, cls in predicts.tolist():
+                predict_result["boxes"].append(box)
+                predict_result["labels"].append(cls)
+                predict_result["scores"].append(conf)
+        return results
+    return post_process_func
+
+
+OUTPUT_SCHEMA = (
+    "struct<boxes:array<array<float>>, scores:array<float>, "
```
Comment (eddyxu) [inline python/rikai/contrib/torch/transforms/yolov5.py:45]: why boxes and scores are in one struct, while labels are not?

**Comment by eddyxu** [comment]
can we use pytorch hub to download yolov5

https://github.com/ultralytics/yolov5/issues/36

File: python/rikai/contrib/torch/transforms/yolov5.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Any, Callable, Dict
+
+
+__all__ = ["post_processing", "OUTPUT_SCHEMA"]
```
Comment (eddyxu) [inline python/rikai/contrib/torch/transforms/yolov5.py:18]: don't need preprocesing?

File: python/rikai/contrib/torch/transforms/yolov5.py (hunk 1 above)
Comment (changhiskhan) [reply python/rikai/contrib/torch/transforms/yolov5.py:45]: multiline string :)

File: python/rikai/contrib/torch/transforms/yolov5.py (hunk 2 above)
Comment (changhiskhan) [reply python/rikai/contrib/torch/transforms/yolov5.py:18]: yeah, the yolo interface can take uri numpy pil etc

Comment (eddyxu) [reply python/rikai/contrib/torch/transforms/yolov5.py:18]: nice

File: python/tests/spark/sql/codegen/test_mlflow_registry.py [hunk 3]
```diff
@@ -82,8 +82,10 @@ def mlflow_client(
             registered_model_name="vanilla-mlflow-no-tags",
         )
 
-    spark.conf.set("rikai.sql.ml.registry.mlflow.tracking_uri", tracking_uri)
-    return mlflow.tracking.MlflowClient(tracking_uri)
+
+@pytest.fixture(scope="session")
+def yolo_model():
+    return torch.hub.load("ultralytics/yolov5", "yolov5s")
```
Comment (changhiskhan) [inline python/tests/spark/sql/codegen/test_mlflow_registry.py:88]: @eddyxu this is what's failing. The issue if you look at yolov5 code (https://github.com/ultralytics/yolov5) is that it's not packaged like a python module so when yolov5 imports from modules in the same repo, it fails in the unit tests.

Comment (changhiskhan) [reply python/tests/spark/sql/codegen/test_mlflow_registry.py:88]:
```
________________________________________________ ERROR at setup of test_mlflow_yolo ________________________________________________

    @pytest.fixture(scope="session")
    def yolo_model():
>       return torch.hub.load("ultralytics/yolov5", "yolov5s")

tests/spark/sql/codegen/test_mlflow_registry.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/chang/miniconda3/envs/eto/lib/python3.7/site-packages/torch/hub.py:339: in load
    model = _load_local(repo_or_dir, model, *args, **kwargs)
/home/chang/miniconda3/envs/eto/lib/python3.7/site-packages/torch/hub.py:365: in _load_local
    hub_module = import_module(MODULE_HUBCONF, hubconf_path)
/home/chang/miniconda3/envs/eto/lib/python3.7/site-packages/torch/hub.py:74: in import_module
    spec.loader.exec_module(module)
<frozen importlib._bootstrap_external>:728: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
/home/chang/.cache/torch/hub/ultralytics_yolov5_master/hubconf.py:12: in <module>
    from models.yolo import Model
/home/chang/.cache/torch/hub/ultralytics_yolov5_master/models/yolo.py:11: in <module>
    from models.common import *
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    import math
    from copy import copy
    from pathlib import Path
    
    import numpy as np
    import pandas as pd
    import requests
    import torch
    import torch.nn as nn
    from PIL import Image
    from torch.cuda import amp
    
>   from utils.datasets import letterbox
E   ModuleNotFoundError: No module named 'utils.datasets'; 'utils' is not a package

/home/chang/.cache/torch/hub/ultralytics_yolov5_master/models/common.py:15: ModuleNotFoundError
```

**Review by eddyxu — APPROVED** [review]

**Pull request merged**